go/release_handler.go

```diff
@@ -0,0 +1,384 @@
+
+		h.m.Lock()
+		defer h.m.Unlock()
+
+		_, err = h.updateReleasedCobraDocs(ctx, client, releaseMeta, version)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// TODO: refactor out shared code between here and synchronizeCobraDocs()
+func (h *ReleaseHandler) updateReleasedCobraDocs(
+	ctx context.Context,
+	client *github.Client,
+	releaseMeta *releaseMetadata,
+	version semver.Version,
+) (*github.PullRequest, error) {
+	vitess := git.NewRepo(
+		releaseMeta.repoOwner,
+		"vitess",
+	).WithLocalDir(filepath.Join(h.Workdir(), "vitess"))
+	website := git.NewRepo(
+		releaseMeta.repoOwner,
+		"website",
+	).WithLocalDir(filepath.Join(h.Workdir(), "website"))
+
+	prs, err := website.ListPRs(ctx, client, github.PullRequestListOptions{
+		State:     "open",
+		Head:      "update-release-cobradocs-for-",
+		Base:      "prod",
+		Sort:      "created",
+		Direction: "desc",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	branch := "prod"
+	baseRepo := website
+	op := "update release cobradocs"
+
+	for _, pr := range prs {
+		if pr.GetUser().GetLogin() != h.botLogin {
+			continue
+		}
+
+		// Most recent PR created by the bot. Base a new PR off of it.
+		head := pr.GetHead()
+
+		branch = head.GetRef()
+		repo := head.GetRepo()
+		baseRepo = git.NewRepo(repo.GetOwner().GetLogin(), repo.GetName())
+
+		zerolog.DefaultContextLogger.Debug().Msgf("using existing PR #%d (%s/%s:%s)", pr.GetNumber(), baseRepo.Owner, baseRepo.Name, branch)
+		break
+	}
+
+	baseRef, _, err := client.Git.GetRef(ctx, baseRepo.Owner, baseRepo.Name, "heads/"+branch)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to fetch %s ref for repository %s/%s to %s for %s", branch, baseRepo.Owner, baseRepo.Name, op, version.String())
+	}
+
+	newBranch := fmt.Sprintf("update-release-cobradocs-for-%s", version.String())
+	_, err = website.CreateBranch(ctx, client, baseRef, newBranch)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create git ref %s for repository %s/%s to %s for %s", newBranch, website.Owner, website.Name, op, version.String())
+	}
+
+	if err := setupRepo(ctx, vitess, fmt.Sprintf("%s for %s", op, version.String())); err != nil {
+		return nil, err
+	}
+
+	if err := vitess.FetchRef(ctx, "origin", "--tags"); err != nil {
+		return nil, errors.Wrapf(err, "Failed to fetch tags in repository %s/%s to %s for %s", vitess.Owner, vitess.Name, op, version.String())
+	}
+
+	if err := setupRepo(ctx, website, fmt.Sprintf("%s for %s", op, version.String())); err != nil {
+		return nil, err
+	}
+
+	// Checkout the new branch we created.
+	if err := website.Checkout(ctx, newBranch); err != nil {
+		return nil, errors.Wrapf(err, "Failed to checkout repository %s/%s to branch %s to %s for %s", website.Owner, website.Name, newBranch, op, version.String())
+	}
+
+	awk, err := shell.NewContext(ctx,
+		"awk",
+		"-F\"",
+		"-e",
+		`$0 ~ /COBRADOC_VERSION_PAIRS="?([^"])"?/ { printf $2 }`,
+		"Makefile",
+	).InDir(website.LocalDir).Output()
```

Comment:
Adding a comment here to explain what this does would be useful

Reply:
https://github.com/vitessio/vitess-bot/pull/66/commits/2436db7072731fae19e8df3c37e6efc223e56440